src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts

```diff
@@ -135,6 +152,16 @@ module DotNet {
      * @param argsJson JSON representation of arguments to pass to the method.
      */
     beginInvokeDotNetFromJS(callId: number, assemblyName: string | null, methodIdentifier: string, dotNetObjectId: number | null, argsJson: string): void;
+
+    /**
+     * Invoked by the runtime to begin an asynchronous call to a .NET method.
+     *
+     * @param callId A value identifying the asynchronous operation. This value should be passed back in a later call from .NET to JS.
+     * @param succeded Whether the operation succeeded or not.
+     * @param resultOrError The result or the error from the async operation.
+     * @param argReplacer A function that replaces [DotNetObject] references with the appropriate value during serialization.
```

Comment:
This layering doesn't look right. Why don't we pass a pre-serialized value to `endInvokeDotNetFromJS`? That way the implementer doesn't need to know about the concept of `argReplacer` and can't accidentally do the serialization incorrectly.

Reply:
I think I tried that, I don't remember the issue I ran into, I'll give it another try. I think it is because I wanted access to the raw error on the server to log it.

I can pass in callId, succeded, and Json.stringify([callId, succeeded, resultOrError], argReplacer) as arguments to this method, but then on the server I don't have a good way to simply access the error message without parsing the payload, which is what I wanted to avoid.

I think we might be ok simply logging serializedArgs on the sever, if that's the case we can simply pass in the serialized args here directly..

Reply:
> I think we might be ok simply logging serializedArgs on the sever,

👍

Reply:
done